tests/functional/test_show.py

```diff
@@ -79,6 +79,30 @@ def test_find_package_not_found():
     assert len(list(result)) == 0
 
 
+def test_report_single_not_found(script):
+    """
+    Test that a single unmatched package is reported not found.
+    This should report an error as there no result to print.
+    """
+    result = script.pip('show', 'abcd3', allow_stderr_warning=True,
+                        expect_error=True)
```

Comment:
This argument can be dropped, IIUC.

Reply:
Oh, wrong line. I meant allow_stderr_warning can be omitted.

Reply:
This was included as a belt and braces following some problems passing tests, at the moment I agree that the expect_error sets allow_stderr_warning for you if it is missing but that may change in the future.

Reply:
I don’t think that will be changing, FYI..

Reply:
I was thinking of the discussion on restructuring logging, `expect_error` is checking the return code, `allow_stderr_warning` _may_ not be always implied. In this case we are actually looking for the warning but the `print_results` gives an error if there are __no__ matches as should be the case in this test.

Reply:
I would code it to how the code is currently written. Otherwise, the code will have an unneeded argument. If things change in the future (which would need its own discussion, but not here), then each call site will need to be revisited anyways.

Reply:
But if you want, you can add a code comment with your findings re: stderr if it is useful info.

Reply:
Done & pushed!